### PR TITLE
Do not allow instructors to deselect both modes (online code editor and offline IDE) in programming exercises

### LINKED ISSUE #2139: Complaints for file upload exercise possible before result is shown
### Describe the bug
Results for a file upload exercises are only shown to the student if the assessment due date has passed or no assessment due date was set. On the other hand a student is able to complain or request more feedback as soon as the exercise due date has passed.

#### To Reproduce

1. As a instructor, create a file upload exercise. Set due date and assessment due date
2. As a student, participate in the file upload exercise
3. As a tutor, assess the file upload exercise after the due date
4. As the student, check your the result before the exercise due date
--> No result is shown
5. You can complain or request more feedback

#### Expected behavior
You should only be able to complain, if the result is shown. Either make the results available as soon as an assessment is submitted or hide the complaint/more feedback button until the assessment due date.

### PR DESCRIPTION
**Pull request by kriststoja**
Fixes #2139 

### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context
When creatin a new programming exercise (or editing it), at least one of "Allow Offline IDE" and "Allow Online Editor" must be selected

### Description
Added a validation to prevent deselecting both button "Allow Offline IDE" and "Allow Online Editor". If both of them are deselected, an info icon pops up with a message explaining the problem and both the text and the icon are in red color. Also the "generate" button will be disabled until at least one of the options is selected.

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Administration
3. Select a course (if none possible generate a new one) and go to exercises.
4. Press "Generate a new programming exercise"
5. At the end of the form are the two options"Allow Offline IDE" and "Allow Online Editor" . Try selecting and deselecting both of them and see if the functionality fullfills the conditions. 
6. Test the disabling and enabling of the "generate" button by filling all the mandatory fields. 

### Test Coverage
programming-exercise-update.component.html 100%

### Screenshots
image
image
image

### CODE REVIEW

**Review by airelawaleria — APPROVED** [review]
Tested in the testing session. Approve

**Review by simonzilker — APPROVED** [review]
Tested in testing session with @EneaGore @airelawaleria @marvinchimeka . Approve!

**Review by marvinchimeka — DISMISSED** [review]
Testing in testing session on ts1. Approve

**Review by EneaGore — DISMISSED** [review]
Tested in testing session on ts1, approve.

**Review by JohannesWeiss — APPROVED** [review]
Works, approve!

**Review by valeriebucher — DISMISSED** [review]
Tested on ts3, works fine!

**Review by LudwigWeide — APPROVED** [review]
Tested on TS1. Works as intended.

File: src/main/webapp/i18n/de/programmingExercise.json [hunk 1]
```diff
@@ -40,9 +40,15 @@
             "repositories": "Repositories",
             "buildplans": "Build Pläne",
             "publishBuildPlanUrl": "Build Plan veröffentlichen",
-            "allowOnlineEditor": "Online Editor erlauben",
+            "allowOnlineEditor": {
+                "title": "Online Editor erlauben",
+                "alert": "Es muss mindestens eine zwischen Offline IDE und Online Editor ausgewählt sein"
```
Comment (krusche) [inline src/main/webapp/i18n/de/programmingExercise.json:45]:
```suggestion
                "alert": "Es muss mindestens eine Option (Offline IDE oder Online Editor) ausgewählt sein"
```

File: src/main/webapp/i18n/de/programmingExercise.json [hunk 2]
```diff
@@ -40,9 +40,15 @@
...
+            },
             "onlineEditor": "Online Editor",
-            "allowOfflineIde": "Offline IDE erlauben",
+            "allowOfflineIde": {
+                "title": "Offline IDE erlauben",
+                "alert": "Es muss mindestens eine zwischen Offline IDE und Online Editor ausgewählt sein"
```
Comment (krusche) [inline src/main/webapp/i18n/de/programmingExercise.json:50]:
```suggestion
                "alert": "Es muss mindestens eine Option (Offline IDE oder Online Editor) ausgewählt sein"
```

File: src/main/webapp/i18n/en/programmingExercise.json [hunk 3]
```diff
@@ -40,9 +40,15 @@
             "repositories": "Repositories",
             "buildplans": "Build Plans",
             "publishBuildPlanUrl": "Publish Build Plan",
-            "allowOnlineEditor": "Allow Online Editor",
+            "allowOnlineEditor": {
+                "title": "Allow Online Editor",
+                "alert": "At least one between Offline IDE and Online Editor must be selected"
```
Comment (krusche) [inline src/main/webapp/i18n/en/programmingExercise.json:45]:
```suggestion
                "alert": "At least one option (Offline IDE or Online Editor) must be selected"
```

File: src/main/webapp/i18n/en/programmingExercise.json [hunk 4]
```diff
@@ -40,9 +40,15 @@
...
+            },
             "onlineEditor": "Online Editor",
-            "allowOfflineIde": "Allow Offline IDE",
+            "allowOfflineIde": {
+                "title": "Allow Offline IDE",
+                "alert": "At least one between Offline IDE and Online Editor must be selected"
```
Comment (krusche) [inline src/main/webapp/i18n/en/programmingExercise.json:50]:
```suggestion
                "alert": "At least one option (Offline IDE or Online Editor) must be selected"
```

**Review by krusche — CHANGES_REQUESTED** [review]
Changes look good, please adapt the translations

**Comment by kriststoja** [comment]
> Changes look good, please adapt the translations

Done

**Review by FuchsDominik — APPROVED** [review]
Code changes looks good :)

**Review by airelawaleria — APPROVED** [review]
Re-approve

**Review by JohannesWeiss — APPROVED** [review]
Code looks good to me, re-approve.

**Review by simonzilker — APPROVED** [review]
re-approve!

**Review by LudwigWeide — APPROVED** [review]
Re-approve.

**Review by marvinchimeka — APPROVED** [review]

**Review by JannisBittlmayer — APPROVED** [review]
Tested on TS2, everything works as intended

**Review by EneaGore — APPROVED** [review]
Re-approve

**Review by valeriebucher — APPROVED** [review]
Reapprove.

**Review by tiffca — APPROVED** [review]
Works as expected, nice change!

**Review by MaisiKoleni — APPROVED** [review]
Code looks good to me.

**Review by nicolasruscher — APPROVED** [review]

**Review by krusche — APPROVED** [review]

**Pull request merged**